From e02546ce5945379006792ade1974ac699535c4fe Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Thu, 5 Dec 2024 18:20:00 +0100
Subject: [PATCH 1408/1695] drivers: stm32_rng: fix configuration field
 definition

Invert RNG_CR_CONFIG1 and RNG_CR_CONFIG3 definitions to fit with
Reference Manual content.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2c8f07f1ec85ff475c0b1a1dea4108f10150ed6a drivers: stm32_rng: fix configuration field definition]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I252b67e76866872a8b0ca061b75ce45c45382811
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423265
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 core/drivers/stm32_rng.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index 80ae7edba..04d232d07 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -39,16 +39,16 @@
 #define RNG_CR_RNGEN		BIT(2)
 #define RNG_CR_IE		BIT(3)
 #define RNG_CR_CED		BIT(5)
-#define RNG_CR_CONFIG1		GENMASK_32(11, 8)
+#define RNG_CR_CONFIG3		GENMASK_32(11, 8)
 #define RNG_CR_NISTC		BIT(12)
 #define RNG_CR_POWER_OPTIM	BIT(13)
 #define RNG_CR_CONFIG2		GENMASK_32(15, 13)
 #define RNG_CR_CLKDIV		GENMASK_32(19, 16)
 #define RNG_CR_CLKDIV_SHIFT	U(16)
-#define RNG_CR_CONFIG3		GENMASK_32(25, 20)
+#define RNG_CR_CONFIG1		GENMASK_32(25, 20)
 #define RNG_CR_CONDRST		BIT(30)
-#define RNG_CR_ENTROPY_SRC_MASK	(RNG_CR_CONFIG1 | RNG_CR_NISTC | \
-				 RNG_CR_CONFIG2 | RNG_CR_CONFIG3)
+#define RNG_CR_ENTROPY_SRC_MASK	(RNG_CR_CONFIG3 | RNG_CR_NISTC | \
+				 RNG_CR_CONFIG2 | RNG_CR_CONFIG1)
 
 #define RNG_SR_DRDY		BIT(0)
 #define RNG_SR_CECS		BIT(1)
-- 
2.47.2

